### ISSUE TITLE
Node 20.x Initial installation fails with apt Hash Sum mismatch

### ISSUE THREAD

**Issue by staehle** [nodesource/distributions#1572]
Hi, I haven't used node before, so I was pointed to this project to install it. My machine is running Ubuntu 22.04. Following the install instructions, I get this error:

```
jrs@cv-penguin $ curl -fsSL https://deb.nodesource.com/setup_20.x | sudo -E bash -
[sudo] password for jrs: 

## Installing the NodeSource Node.js 20.x repo...


## Populating apt-get cache...

+ apt-get update
Get:1 https://apt.releases.hashicorp.com jammy InRelease [12.9 kB]
Get:2 https://apt.releases.hashicorp.com jammy/main amd64 Packages [86.0 kB]                                                                                                        
Get:3 https://apt.releases.hashicorp.com jammy/main i386 Packages [33.3 kB]                                                                                                                 
Hit:4 http://us.archive.ubuntu.com/ubuntu jammy InRelease                                                                                                                                        
Get:5 http://us.archive.ubuntu.com/ubuntu jammy-updates InRelease [119 kB]                                                                    
Get:6 http://security.ubuntu.com/ubuntu jammy-security InRelease [110 kB]                                                                                
Get:7 http://us.archive.ubuntu.com/ubuntu jammy-backports InRelease [108 kB]                                                                             
Get:8 https://ppa.launchpadcontent.net/mozillateam/ppa/ubuntu jammy InRelease [23.8 kB]   
Get:9 http://us.archive.ubuntu.com/ubuntu jammy-updates/main amd64 Packages [662 kB]  
Get:10 http://us.archive.ubuntu.com/ubuntu jammy-updates/main i386 Packages [412 kB]            
Get:11 http://us.archive.ubuntu.com/ubuntu jammy-updates/main Translation-en [179 kB]     
Get:12 http://us.archive.ubuntu.com/ubuntu jammy-updates/main amd64 DEP-11 Metadata [102 kB]     
Get:13 http://us.archive.ubuntu.com/ubuntu jammy-updates/main DEP-11 48x48 Icons [33.0 kB]        
Get:14 http://us.archive.ubuntu.com/ubuntu jammy-updates/main DEP-11 64x64 Icons [51.3 kB]
Get:15 http://us.archive.ubuntu.com/ubuntu jammy-updates/main amd64 c-n-f Metadata [14.7 kB]
Get:16 http://us.archive.ubuntu.com/ubuntu jammy-updates/universe amd64 Packages [907 kB]
Get:17 http://security.ubuntu.com/ubuntu jammy-security/main amd64 Packages [442 kB]
Get:18 http://us.archive.ubuntu.com/ubuntu jammy-updates/universe i386 Packages [616 kB]
Get:19 http://security.ubuntu.com/ubuntu jammy-security/main i386 Packages [227 kB]
Get:20 http://us.archive.ubuntu.com/ubuntu jammy-updates/universe Translation-en [187 kB]
Get:21 http://us.archive.ubuntu.com/ubuntu jammy-updates/universe amd64 DEP-11 Metadata [269 kB]
Get:22 http://security.ubuntu.com/ubuntu jammy-security/main Translation-en [120 kB]                     
Get:23 http://us.archive.ubuntu.com/ubuntu jammy-updates/universe amd64 c-n-f Metadata [19.5 kB]       
Get:24 http://us.archive.ubuntu.com/ubuntu jammy-updates/multiverse amd64 DEP-11 Metadata [940 B]         
Get:25 http://us.archive.ubuntu.com/ubuntu jammy-backports/main amd64 DEP-11 Metadata [7,976 B]
Get:26 http://us.archive.ubuntu.com/ubuntu jammy-backports/universe amd64 DEP-11 Metadata [16.8 kB]         
Get:27 http://security.ubuntu.com/ubuntu jammy-security/main amd64 DEP-11 Metadata [41.6 kB]                
Get:28 http://security.ubuntu.com/ubuntu jammy-security/main amd64 c-n-f Metadata [10.0 kB]
Get:29 http://security.ubuntu.com/ubuntu jammy-security/universe amd64 Packages [729 kB]
Get:30 http://security.ubuntu.com/ubuntu jammy-security/universe i386 Packages [532 kB]
Get:31 http://security.ubuntu.com/ubuntu jammy-security/universe Translation-en [128 kB]
Get:32 http://security.ubuntu.com/ubuntu jammy-security/universe amd64 DEP-11 Metadata [19.1 kB]
Get:33 http://security.ubuntu.com/ubuntu jammy-security/universe amd64 c-n-f Metadata [15.3 kB]
Fetched 6,235 kB in 2s (3,901 kB/s)                                
Reading package lists... Done

## Confirming "jammy" is supported...

+ curl -sLf -o /dev/null 'https://deb.nodesource.com/node_20.x/dists/jammy/Release'

## Adding the NodeSource signing key to your keyring...

+ curl -s https://deb.nodesource.com/gpgkey/nodesource.gpg.key | gpg --dearmor | tee /usr/share/keyrings/nodesource.gpg >/dev/null
gpg: WARNING: unsafe ownership on homedir '/home/jrs/.gnupg'

## Creating apt sources list file for the NodeSource Node.js 20.x repo...

+ echo 'deb [signed-by=/usr/share/keyrings/nodesource.gpg] https://deb.nodesource.com/node_20.x jammy main' > /etc/apt/sources.list.d/nodesource.list
+ echo 'deb-src [signed-by=/usr/share/keyrings/nodesource.gpg] https://deb.nodesource.com/node_20.x jammy main' >> /etc/apt/sources.list.d/nodesource.list

## Running `apt-get update` for you...

+ apt-get update
Hit:1 https://apt.releases.hashicorp.com jammy InRelease
Get:2 https://deb.nodesource.com/node_20.x jammy InRelease [4,563 B]                                                                                                                                                                        
Hit:3 http://us.archive.ubuntu.com/ubuntu jammy InRelease                                                                                                                   
Get:4 https://deb.nodesource.com/node_20.x jammy/main amd64 Packages [776 B]                                                                   
Err:4 https://deb.nodesource.com/node_20.x jammy/main amd64 Packages                                                                 
  Hash Sum mismatch
  Hashes of expected file:
   - Filesize:776 [weak]
   - SHA256:968f67bce6165b9c311cd55c1e546befaa2f27e83c5e7184ed2abdf89d55568f
   - SHA1:88e150f7ba75015ae8e2875b65af41534c8bf3cc [weak]
   - MD5Sum:bf9809a56ad11691d0197775a5abee90 [weak]
  Hashes of received file:
   - SHA256:b8ca63ac4fbe9dad6950b850a9258db453c68c7b1c60f25457b9709684154e47
   - SHA1:35c0928a4089a3064e90b1c3cd0eb3f90d96983f [weak]
   - MD5Sum:333f082eb4c6371f533f47acb4774153 [weak]
   - Filesize:776 [weak]
  Last modification reported: Tue, 16 May 2023 15:52:07 +0000
  Release file created at: Thu, 04 May 2023 14:47:55 +0000
Hit:5 http://us.archive.ubuntu.com/ubuntu jammy-updates InRelease                                                                    
Hit:6 http://us.archive.ubuntu.com/ubuntu jammy-backports InRelease                                        
Hit:7 http://security.ubuntu.com/ubuntu jammy-security InRelease                                           
Hit:8 https://ppa.launchpadcontent.net/mozillateam/ppa/ubuntu jammy InRelease        
Fetched 5,339 B in 1s (10.4 kB/s)
Reading package lists... Done
E: Failed to fetch https://deb.nodesource.com/node_20.x/dists/jammy/main/binary-amd64/Packages.gz  Hash Sum mismatch
   Hashes of expected file:
    - Filesize:776 [weak]
    - SHA256:968f67bce6165b9c311cd55c1e546befaa2f27e83c5e7184ed2abdf89d55568f
    - SHA1:88e150f7ba75015ae8e2875b65af41534c8bf3cc [weak]
    - MD5Sum:bf9809a56ad11691d0197775a5abee90 [weak]
   Hashes of received file:
    - SHA256:b8ca63ac4fbe9dad6950b850a9258db453c68c7b1c60f25457b9709684154e47
    - SHA1:35c0928a4089a3064e90b1c3cd0eb3f90d96983f [weak]
    - MD5Sum:333f082eb4c6371f533f47acb4774153 [weak]
    - Filesize:776 [weak]
   Last modification reported: Tue, 16 May 2023 15:52:07 +0000
   Release file created at: Thu, 04 May 2023 14:47:55 +0000
E: Some index files failed to download. They have been ignored, or old ones used instead.
Error executing command, exiting
jrs@cv-penguin $ 
```

**Comment by riosje** [comment]
Hi @staehle 
I was not able to re-produce your error, but you may want to try this

```SHELL
sudo apt-get clean
sudo apt-get update
sudo apt-get install -y nodejs
```

**Comment by staehle** [comment]
Hey thanks for looking at this @riosje!

When I attempted to run `sudo apt update` yesterday on this same workstation, I kept getting the same hash error, so I just dropped it for the day.  But this morning I tried it again, and the issue seems to have resolved itself (which is always wonderfully discomforting thing haha)! I can now run `sudo apt install nodejs` successfully.

So... I can't reproduce it again either! Maybe this was just a one-off bug with GPG? ¯\_(ツ)_/¯

Thanks anyways!
